core/src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -267,6 +279,62 @@ public MetadataSnapshot getMetadata(IndexCommit commit) throws IOException {
         }
     }
 
+    public IndexCommit findLastCommitBellowSeqNo(final long seqNo) throws IOException {
+        ensureOpen();
+        failIfCorrupted();
+        metadataLock.readLock().lock();
+        IndexCommit lastCommit = null;
+        long commitMaxSeqNo = UNASSIGNED_SEQ_NO;
+        try {
+            for (IndexCommit commit : DirectoryReader.listCommits(directory)) {
```

Comment:
can you commetn that `DirectoryReader.listCommits` guarantees commits in ascending order